embedded/embedded-impl/src/main/java/org/mule/runtime/module/embedded/impl/EmbeddedController.java

```diff
@@ -65,7 +65,9 @@ public synchronized void deployApplication(byte[] serializedArtifactConfiguratio
       if (artifactConfiguration.getDeploymentConfiguration().enableTestDependencies()) {
         setProperty(ADD_TEST_DEPENDENCIES_KEY, "true");
       }
+      muleContainer.getDeploymentService().getLock().lock();
       muleContainer.getDeploymentService().deploy(artifactConfiguration.getArtifactLocation().toURI());
+      muleContainer.getDeploymentService().getLock().unlock();
```

Comment:
this has to be done inside a finally block, otherwise the lock will be keep locked by this thread